### PR TITLE
[Omn-626] - API to edit title and/or description

### PR DESCRIPTION
**Pull request by gitstart-omnivore**
## Spec Checklist
- [ ] Are Api endpoints ready and properly documented ?
- [ ] Added video describing your plan of action ?

## Description

Added a new graphql API updatePage that takes a page id, and optionally a title and a description. And added a test for it.

## House Keeping
- [ ] Added video describing your plan of action ?
- [ ] Added Loom video ?
- [x] Is Deployment Link passing ?
- [x] Have you assinged PR to yourself ?
- [x] Is Github action passing ?
- [x] Updated the appropriate tag ?

## Ticket link (if applicable)

[Task Link](https://app.gitstart.com/clients/omnivore/tickets/OMN-626)

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Others

## Checklist:

- [ ] I have test my code on different browsers and devices to make sure it is cross-browser compatible (both desktop and mobile).
- [ ] I have confirmed the payload for integrated API matches params on API docs

## Remarks

- [ ] I added a new package to achieve this task.
- [ ] I have updated package.json

### CODE REVIEW

**Comment by jacksonh** [comment]
@gitstart-omnivore checkout this file for a good example of how we like to do API tests: https://github.com/omnivore-app/omnivore/blob/main/packages/api/test/resolvers/labels.test.ts

**Comment by gitstart-omnivore** [comment]
@jacksonh This PR is ready for review.

**Comment by jacksonh** [comment]
@sywhb do you mind reviewing this one?

**Comment by sywhb** [comment]
> @sywhb do you mind reviewing this one?

Sure. Sorry I didn't see this one coming out

**Comment by sywhb** [comment]
@jacksonh @gitstart-omnivore Overall it looks good and I have made a few comments here

File: packages/api/src/resolvers/update/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    models,
+    claims: { uid },
+  } = ctx
+
+  const user = userDataToUser(await models.user.get(uid))
+  if (!user) {
+    return { errorCodes: [UpdatePageErrorCode.Unauthorized] }
+  }
+
+  const pageData = {
```
[the remark this answers is not in the document]
Comment (gitstart-omnivore) [reply packages/api/src/resolvers/update/index.ts:25]: @jacksonh @sywhb Do we also want to make it possible to set title as an empty string?

**Comment by gitstart-omnivore** [comment]
@sywhb We have made the changes, can you please take a look again. Thanks!

File: packages/api/src/resolvers/update/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sywhb) [reply packages/api/src/resolvers/update/index.ts:38]: I think `title` is compulsory

[the remark this answers is not in the document]
Comment (jacksonh) [reply packages/api/src/resolvers/update/index.ts:38]: Yeah, we need to have a title normally, so we shouldn't allow the title to be empty, but an empty description is OK.

File: packages/api/src/resolvers/update/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    id: input.pageId,
+    title: input.title || undefined,
+    description: input.description || undefined,
+  }
+
+  const updateResult = await updatePage(input.pageId, pageData, { ...ctx, uid })
```
[the remark this answers is not in the document]
Comment (sywhb) [reply packages/api/src/resolvers/update/index.ts:44]: I guess we should verify it before updating

[the remark this answers is not in the document]
Comment (sywhb) [reply packages/api/src/resolvers/update/index.ts:44]: @gitstart-omnivore

[the remark this answers is not in the document]
Comment (gitstart-omnivore) [reply packages/api/src/resolvers/update/index.ts:44]: @sywhb  Is it not enough to check the userId of the page against the id of the user who is making the request? We have added this validation.

[the remark this answers is not in the document]
Comment (sywhb) [reply packages/api/src/resolvers/update/index.ts:44]: Hey sorry didn't see your change.
Yes, it is enough

**Review by sywhb — APPROVED** [review]

**Pull request merged**